### PR TITLE
[UNMERGED] #151243725 GET all contribution types

### PR DESCRIPTION
**Pull request by andela-ieyo**
#### What does this PR do? ####
  - create endpoint to retrieve all contribution types
  - create view function to handle GET all contribution types

#### How should this be manually tested? ####
    At the root of the project directory run the following command:
    $ DB_TYPE=sqlite pytest -v OR $ DB_TYPE=sqlite pytest

#### What are the relevant PT stories? ####
           [151243725]

### CODE REVIEW

File: api/views/contributions.py [hunk 1]
```diff
@@ -587,3 +587,18 @@ def get(self, notif_id=None):
             ))
         response.status_code = 200
         return response
+
+
+class ContributionTypeResource(Resource):
+
+    @token_required
+    def get(self):
```
Comment (andela-milesanmi) [inline api/views/contributions.py]: @andela-ieyo, Nice job. I think it will be good to catch errors also.

**Comment by wanjiku-carol** [comment]
Its good. Maybe also add the tests for the end point. `test_contributiontypes.py`?

File: api/views/contributions.py (hunk 1 above)
Comment (andela-ieyo) [reply api/views/contributions.py]: @mai If line 596 fails, the base error handling function on `main.py`gets called, so i don't see a need for another error check....this is the same approach adopted for a query like the one on line 596 across the codebase. I stand to be corrected though

File: migrations/versions/550bc502b5f3_.py [hunk 2]
```diff
@@ -1,74 +0,0 @@
-"""empty message
-
```
Comment (LarryWachira) [inline migrations/versions/550bc502b5f3_.py]: Did you migrate @andela-ieyo?

**Comment by LarryWachira** [comment]
Nicely done. Please add tests for the endpoint.

File: migrations/versions/550bc502b5f3_.py (hunk 2 above)
Comment (andela-ieyo) [reply migrations/versions/550bc502b5f3_.py]: yes, i did

File: test/api/views/test_contributions.py [hunk 3]
```diff
@@ -105,7 +113,14 @@ def test_view_all_contributions(self):
         response_data = json.loads(response.data)
         self.assertEqual(len(response_data['data']['contributions']), 0)
         self.assert200(response)
-                        
+
+    def test_view_all_contribution_types(self):
+        response = self.client.get('/api/v1/contribution/types',
+                                    headers=self.regular_header)
+        response_data = json.loads(response.data)
+        self.assertEqual(len(response_data['data']['contributionTypes']), 3)
```
Comment (LarryWachira) [inline test/api/views/test_contributions.py]: Consider making a query and count on the contribution types table which you can then use to compare with the response contribution types count.

File: api/views/contributions.py (hunk 1 above)
Comment (andela-milesanmi) [reply api/views/contributions.py]: @andela-ieyo Thats right. Thanks for the clarification...

File: test/api/views/test_contributions.py (hunk 3 above)
Comment (andela-ieyo) [reply test/api/views/test_contributions.py]: done.

**Comment by LarryWachira** [comment]
Please rebase @andela-ieyo

**Comment by LarryWachira** [comment]
LGTM. But please wait while we confirm how to handle migrations officially.

**Comment by andela-oowonikoko** [comment]
LGTM

**Comment by LarryWachira** [comment]
Do you mind moving the work here over to your tags PR @andela-ieyo? You also have a failing test probably due to the change in the URL. Since tags is a minor change I don't see the need for it to have its own commit on develop.

**Pull request closed without merging**